clients/src/main/java/org/apache/kafka/common/ApiKey.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common;
+
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+import org.apache.kafka.common.protocol.Protocol;
+
+/**
+ * Identifiers for Kafka APIs.
+ *
+ * This is a public API for both Kafka clients and internals.  Metadata which is not intended to be
+ * exposed to end-users should go in #{org.apache.kafka.common.protocol.ApiKeys} rather than here.
+ *
+ * Each ApiKey represents an API which Kafka brokers understand.  In order to maintain
+ * compatibility, we can add more APIs to the list, but not remove or reorder existing APIs.
+ */
+public enum ApiKey {
+    PRODUCE("Produce"),
+    FETCH("Fetch"),
+    LIST_OFFSETS("Offsets"),
+    METADATA("Metadata"),
+    LEADER_AND_ISR("LeaderAndIsr"),
+    STOP_REPLICA("StopReplica"),
+    UPDATE_METADATA_KEY("UpdateMetadata"),
+    CONTROLLED_SHUTDOWN_KEY("ControlledShutdown"),
+    OFFSET_COMMIT("OffsetCommit"),
+    OFFSET_FETCH("OffsetFetch"),
+    FIND_COORDINATOR("FindCoordinator"),
+    JOIN_GROUP("JoinGroup"),
+    HEARTBEAT("Heartbeat"),
+    LEAVE_GROUP("LeaveGroup"),
+    SYNC_GROUP("SyncGroup"),
+    DESCRIBE_GROUPS("DescribeGroups"),
+    LIST_GROUPS("ListGroups"),
+    SASL_HANDSHAKE("SaslHandshake"),
+    API_VERSIONS("ApiVersions"),
+    CREATE_TOPICS("CreateTopics"),
+    DELETE_TOPICS("DeleteTopics"),
+    DELETE_RECORDS("DeleteRecords"),
+    INIT_PRODUCER_ID("InitProducerId"),
+    OFFSET_FOR_LEADER_EPOCH("OffsetForLeaderEpoch"),
+    ADD_PARTITIONS_TO_TXN("AddPartitionsToTxn"),
+    ADD_OFFSETS_TO_TXN("AddOffsetsToTxn"),
+    END_TXN("EndTxn"),
+    WRITE_TXN_MARKERS("WriteTxnMarkers"),
+    TXN_OFFSET_COMMIT("TxnOffsetCommit"),
+    DESCRIBE_ACLS("DescribeAcls"),
+    CREATE_ACLS("CreateAcls"),
+    DELETE_ACLS("DeleteAcls"),
+    DESCRIBE_CONFIGS("DescribeConfigs"),
+    ALTER_CONFIGS("AlterConfigs");
+
+    public static final ApiKey[] VALUES = ApiKey.values();
```

Comment:
It's not ideal that we are exposing the mutable array to users. The findBugs warning that is being suppressed is valid IMO. We could expose an unmodifiable list perhaps?